### PR TITLE
Chaincode invocation response encryption

### PR DESCRIPTION
**Pull request by uraldav**

### CODE REVIEW

File: extensions/encryption/middleware_after.go [hunk 1]
```diff
@@ -4,16 +4,30 @@ import (
 	"github.com/s7techlab/cckit/router"
 )
 
-func EncryptInvokeResponse() router.MiddlewareFunc {
+func responseEncryptor(encryptionRequired bool) router.MiddlewareFunc {
 	return func(pre router.HandlerFunc, pos ...int) router.HandlerFunc {
 		return func(ctx router.Context) (interface{}, error) {
 			res, err := pre(ctx)
 
 			if err != nil || ctx.Handler().Type != router.MethodInvoke {
```
Comment (vitiko) [inline extensions/encryption/middleware_after.go:12]: here we can check presense of encode key in transient map and don't call EncryptWithTransientKey if there is not encode key and encryption not required

File: extensions/encryption/middleware_after.go [hunk 2]
```diff
@@ -4,16 +4,30 @@ import (
 	"github.com/s7techlab/cckit/router"
 )
 
-func EncryptInvokeResponse() router.MiddlewareFunc {
```
Comment (vitiko) [inline extensions/encryption/middleware_after.go:7]: encrypIfEncodeKeyExists - more accurate name

Comment (uraldav) [reply extensions/encryption/middleware_after.go:7]: fixed

File: extensions/encryption/middleware_after.go (hunk 1 above)
Comment (uraldav) [reply extensions/encryption/middleware_after.go:12]: fixed

**Review by vitiko — APPROVED** [review]

**Pull request merged**